jsonnet/dashboards/apps/loki.libsonnet

```diff
@@ -0,0 +1,246 @@
+        .addPanel(ingester_lines { tooltip+: { sort: 2 } }, { x: 16, y: sy+8, w: 8, h: 7 })
+        ;
+   local panel_cache(sy) = row.new('Cache',collapse=true) {gridPos:{x: 0, y:sy, w:24, h:1}}
+        .addPanel(cache_size { tooltip+: { sort: 2 } }, {x: 0, y: sy+1, w: 12, h: 7})
+        .addPanel(cache_fetched_keys { tooltip+: { sort: 2 } }, {x: 12, y: sy+1, w: 12, h: 7})
+        .addPanel(cache_hits_keys { tooltip+: { sort: 2 } }, {x: 0, y: sy+8, w: 12, h: 7})
+        ;
+   local panel_index(sy) = row.new('Index',collapse=true) {gridPos:{x: 0, y:sy, w:24, h:1}}
+        .addPanel(chunk_store_chunks { tooltip+: { sort: 2 } }, {x: 0, y: sy+1, w: 8, h: 7})
+        .addPanel(chunk_store_entries { tooltip+: { sort: 2 } }, {x: 8, y: sy+1, w: 8, h: 7})
+        .addPanel(index_lookups { tooltip+: { sort: 2 } }, {x: 16, y: sy+1, w: 8, h: 7})
+        .addPanel(series_pre_intersection { tooltip+: { sort: 2 } }, {x: 0, y: sy+8, w: 12, h: 7})
+        .addPanel(series_post_intersection { tooltip+: { sort: 2 } }, {x: 12, y: sy+8, w: 12, h: 7})
+        ;
+   local panel_querrier(sy) = row.new('Querier',collapse=true) {gridPos:{x: 0, y:sy, w:24, h:1}}
+        .addPanel(querier_cache_hits { tooltip+: { sort: 2 } }, {x: 0, y: sy+1, w: 8, h: 7})
+        .addPanel(querier_cache_puts { tooltip+: { sort: 2 } }, {x: 8, y: sy+1, w: 8, h: 7})
+        .addPanel(querier_cache_gets { tooltip+: { sort: 2 } }, {x: 16, y: sy+1, w: 8, h: 7})
+        .addPanel(querier_cache_corruptions { tooltip+: { sort: 2 } }, {x: 0, y: sy+8, w: 12, h: 7})
+        .addPanel(querier_cache_errors { tooltip+: { sort: 2 } }, {x: 12, y: sy+8, w: 12, h: 7})
+        ;
+   local panels = [
+        /*Overview*/
+        row.new('Overview') { gridPos: { x: 0, y: 0, w: 24, h: 1 } },
+        version {gridPos: { x: 0, y: 1, w: 4, h: 5 }},
+        msgs { gridPos: { x: 4, y: 1, w: 4, h: 5 } },
+        errors { gridPos: { x: 8, y: 1, w: 4, h: 5 } },
+        panic { gridPos: { x: 12, y: 1, w: 4, h: 5 } },
+        /*Other Panels*/
+        panel_msgs(6),panel_api(14),panel_cache(29),panel_querrier(44),panel_index(59),panel_ingester(74)];
+
+   dashboard.new(
+        'Loki',
+        editable=$._config.grafanaDashboards.editable,
+        graphTooltip=$._config.grafanaDashboards.tooltip,
+        refresh=$._config.grafanaDashboards.refresh,
+        time_from=$._config.grafanaDashboards.time_from,
+        tags=$._config.grafanaDashboards.tags.k8sApps,
+        uid='loki',
+      )
+
+      .addTemplates([
+        $.grafanaTemplates.datasourceTemplate(),
+        $.grafanaTemplates.clusterTemplate('label_values(node_uname_info, cluster)'),
+        $.grafanaTemplates.jobTemplate('label_values(container_cpu_user_seconds_total{cluster=~"$cluster"}, job)'),
+        $.grafanaTemplates.containerTemplate('label_values(container_cpu_user_seconds_total{cluster=~"$cluster", job=~"$job"}, name)'),
+      ])
+      .addPanels(panels),
+   }
+
+}
```

Comment:
Good convention is to have a newline at the end of each file. You can [set Pycharm](https://stackoverflow.com/questions/36043061/automatically-add-newline-on-save-in-pycharm) to do it automatically.